### PR TITLE
Add provinance to the registry entry

### PR DESCRIPTION
**Pull request by bnchrch**
## What
Adds the generation time and root metadata file info to the registry entry

Spun out of #32715 as a stack

closes https://github.com/airbytehq/airbyte-internal-issues/issues/7466

### CODE REVIEW

**Comment by bnchrch** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/37804?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#37804** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/37804?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#37803** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/37803?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#37802** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/37802?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bnchrch and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorRegistryDestinationDefinition.py [hunk 1]
```diff
@@ -79,6 +79,36 @@ class Config:
     ql: Optional[Literal[100, 200, 300, 400, 500, 600]] = None
 
 
+class GitInfo(BaseModel):
+    class Config:
+        extra = Extra.forbid
+
+    commit_sha: Optional[str] = Field(
+        None,
+        description="The git commit sha of the last commit that modified this file. DO NOT DEFINE THIS FIELD IN YOUR SPEC. It will be overwritten by the CI.",
```
Comment (natikgadzhi) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorRegistryDestinationDefinition.py:88]: Nit: should we lift the autogeneration notice up one/two levels and just say that the whole source_file_info probably should not be defined in a spec?

File: airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/utils/object_helpers.py [hunk 2]
```diff
@@ -34,3 +34,21 @@ def f(*args, **kwargs):
         return to_call(*copy.deepcopy(args), **copy.deepcopy(kwargs))
 
     return f
+
+
+def default_none_to_dict(value, key, obj):
+    """Set the value of a key in a dictionary to an empty dictionary if the value is None.
```
Comment (natikgadzhi) [inline airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/utils/object_helpers.py:40]: TY for a nice doc! <3

File: airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/ops/slack.py [hunk 3]
```diff
@@ -29,7 +29,7 @@ def send_slack_message(context: OpExecutionContext, channel: str, message: str,
         channel (str): The channel to send the message to.
         message (str): The message to send.
     """
-    if os.getenv("SLACK_TOKEN"):
+    if os.getenv("SLACK_TOKEN") == "" and os.getenv("SLACK_NOTIFICATIONS_DISABLED") != "true":
```
Comment (natikgadzhi) [inline airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/ops/slack.py:32]: Is the new env var documented somewhere?

**Review by natikgadzhi — APPROVED** [review]
Haven't tested, but the code looks great. Thank you for extensive comments, and for `poetry lock`ing so we get a few vulns fixed while we're at it!

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorRegistrySourceDefinition.py [hunk 4]
```diff
@@ -75,6 +75,36 @@ class Config:
...
+        None,
+        description="The git commit author email of the last commit that modified this file. DO NOT DEFINE THIS FIELD IN YOUR SPEC. It will be overwritten by the CI.",
+    )
+
+
+class SourceFileInfo(BaseModel):
+    metadata_etag: Optional[str] = None
+    metadata_file_path: Optional[str] = None
+    metadata_bucket_name: Optional[str] = None
+    metadata_last_modified: Optional[str] = None
```
Comment (alafanechere) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorRegistrySourceDefinition.py:104]: Nice addition for debugging

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorRegistrySourceDefinition.py [hunk 5]
```diff
@@ -75,6 +75,36 @@ class Config:
...
+    )
+    commit_author_email: Optional[str] = Field(
+        None,
+        description="The git commit author email of the last commit that modified this file. DO NOT DEFINE THIS FIELD IN YOUR SPEC. It will be overwritten by the CI.",
+    )
+
+
+class SourceFileInfo(BaseModel):
+    metadata_etag: Optional[str] = None
+    metadata_file_path: Optional[str] = None
```
Comment (alafanechere) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorRegistrySourceDefinition.py:102]: It's a blob/key path isn't?

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorRegistrySourceDefinition.py [hunk 6]
```diff
@@ -87,6 +117,11 @@ class BreakingChangeScope(BaseModel):
     __root__: StreamBreakingChangeScope = Field(..., description="A scope that can be used to limit the impact of a breaking change.")
 
 
+class GeneratedFields(BaseModel):
```
Comment (alafanechere) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorRegistrySourceDefinition.py:120]: As I said in https://github.com/airbytehq/airbyte/pull/37802 I think all generated fields should stay in the registry, not be metadata fields (which is what you're doing here 👍 )

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorRegistryV0.py [hunk 7]
```diff
@@ -79,6 +79,36 @@ class Config:
...
+        description="The git commit timestamp of the last commit that modified this file. DO NOT DEFINE THIS FIELD IN YOUR SPEC. It will be overwritten by the CI.",
+    )
+    commit_author: Optional[str] = Field(
+        None,
+        description="The git commit author of the last commit that modified this file. DO NOT DEFINE THIS FIELD IN YOUR SPEC. It will be overwritten by the CI.",
+    )
+    commit_author_email: Optional[str] = Field(
+        None,
+        description="The git commit author email of the last commit that modified this file. DO NOT DEFINE THIS FIELD IN YOUR SPEC. It will be overwritten by the CI.",
+    )
```
Comment (alafanechere) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorRegistryV0.py:101]: This is already declared in https://github.com/airbytehq/airbyte/pull/37802
You might need to `gt restack`

**Review by alafanechere — APPROVED** [review]

**Comment by bnchrch** [comment]
### Merge activity

* **May 9, 10:18 AM EDT**: @bnchrch started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/37804).

**Pull request merged**